src/utilities/createRouter.ts

```diff
@@ -45,8 +44,20 @@ export function createRouter<T extends Routes>(routes: T, options: RouterOptions
     Object.assign(route, newRoute)
   }
 
-  const push: RouterPush = async (url, options) => {
-    await navigation.update(url, options)
+  const push: RouterPush<T> = (urlOrRoute, optionsOrParams: Record<string, any> = {}, possiblyOptions: Record<string, any> = {}) => {
+    if (typeof urlOrRoute === 'string') {
+      return navigation.update(urlOrRoute, optionsOrParams)
+    }
+
+    const route = resolved.find(({ name }) => name === urlOrRoute.name)
```

Comment:
This won't work for a nested route. Because `Flattened<Routes>` will produce things `parent.child.grandchild` which won't match a resolve route.

We could probably use `name.split('.').at(-1)` to match the current behavior.

Though matching a resolved route by its name means routes couldn't have the same names like `parentA.child` and `parentB.child`. So far we've said route names do not need to be unique. Idk if that's worth worrying about here.

Reply:
great call, I've got a service that takes Resolved<Route>, combines it's `.matches` to create a string that we can compare the full history. That should retain support for leaf names that aren't unique

Reply:
Great idea